editor/editor_properties.cpp

```diff
@@ -2546,32 +2546,34 @@ void EditorPropertyResource::_menu_option(int p_which) {
 			}
 
 			Object *obj = nullptr;
+			RES res_temp;
 
 			if (ScriptServer::is_global_class(intype)) {
 				obj = ClassDB::instance(ScriptServer::get_global_class_native_base(intype));
 				if (obj) {
+					res_temp = obj;
 					Ref<Script> script = ResourceLoader::load(ScriptServer::get_global_class_path(intype));
 					if (script.is_valid()) {
 						obj->set_script(Variant(script));
 					}
 				}
 			} else {
 				obj = ClassDB::instance(intype);
+				res_temp = obj;
 			}
 
 			if (!obj) {
 				obj = EditorNode::get_editor_data().instance_custom_type(intype, "Resource");
+				res_temp = obj;
```

Comment:
Wouldn't it be equivalent to just do `RES res_temp = obj;` once after `ERR_BREAK(!obj);`?

Reply:
By that time it's already too late, because it's the first case where the early destruction happens: setting the script may involve calling the GDScript constructor and that's where the reference-then-unreference-to-zero thing happens.

Reply:
Also, the other two cases could get by without referencing until later, but I believe that written that way it's more explicit. It's like an idiom: you instance a `Reference`?, then wrap it in a `Ref` ASAP to avoid problems.